posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -183,38 +183,56 @@ object SocketHelpers {
       retArray.toArray
     }
 
+  private def ipStringAddrToSockaddr(
+      ip: String,
+      isV6: Boolean,
+      addr: Ptr[sockaddr])(implicit z: Zone): Boolean = {
+
+    val src = toCString(ip)
+
+    addr.sa_family = { if (isV6) AF_INET6 else AF_INET }.toUShort
+
+    val dst =
+      if (isV6) {
+        addr
+          .asInstanceOf[Ptr[sockaddr_in6]]
+          .sin6_addr
+          .toPtr
+          .asInstanceOf[Ptr[Byte]]
+      } else {
+        addr
+          .asInstanceOf[Ptr[sockaddr_in]]
+          .sin_addr
+          .toPtr
+          .asInstanceOf[Ptr[Byte]]
+      }
+
+    // See error reporting note at bottom of sole caller ipToHost().
+    inet_pton(addr.sa_family.toInt, src, dst) == 1
```

Comment:
Whatever this does could be a clue to the name of the function that calls it. Is this function returning 1 like true in C so that is good?

Reply:
Reworked to be less C and more Scala idiomatic.

Reply:
Re-worked again.  Gave helper method `tailorSockaddr()` a name which, I believe, 
better describes its intended use. 

I tried making that method local rather than private but the results left
the body of `ipToHost()` as nearly as long and confusing as the code this PR
wants to replace.

Reply:
@LeeTibbert This satisfies my ideas - I like the Boolean and the naming better now.